# Gene Expression Levels Using Multiple Linear Regression :dna:
This project involves implementing multiple linear regression from scratch using tensorflow (no build in functions, only tensors and automatic diferentiation), to predict gene expression levels based on various genomic features. The goal is to create a model that can accurately estimate the expression levels of genes given a set of predictors derived from DNA sequences, such as GC content, promoter strength, and the presence of specific motifs.

If you only want to see the code, you can find it [here](MLR_explanation.ipynb).

## Table of Contents
- [Introduction](#introduction)
- [Overview](#overview)
- [Mathemathics behind Linear Regression](#mathemathics-behind-linear-regression)
- [Implementation Details](#implementation-details)
    - [Data Preprocessing](#data-preprocessing)
    - [Model Architecture](#model-architecture)
    - [Loss Function](#loss-function)
    - [Accuracy Function](#accuracy-function)
- [Training the Model](#training-the-network)
- [Results](#results)
- [Usage](#usage)
- [References](#references)

## Introduction
![Training_example](images/training_animation.gif)

## Overview

## Mathemathics behind Linear Regression
I created a `Jupyter Notebook` to explain the mathemathics behind the linear regression and the explained code. You can find the notebook [here](MLR_explanation.ipynb). It's easy to understand and I hope it will help you to understand the mathemathics behind the linear regression.
## Implementation Details
### Data Preprocessing
### Model Architecture
### Loss Function
### Accuracy Function
## Training the Model
## Results
## Usage
## References
